shared/js/media/media_utils.js

```diff
@@ -9,12 +9,20 @@
 var MediaUtils = {
    _: navigator.mozL10n.get,
 
+   _getDateTimeFormat: function() {
+    if (!this._dtf) {
+      this._dtf = new navigator.mozL10n.DateTimeFormat();
+    }
+
+    return this._dtf;
+   },
```

Comment:
Nit: It looks like the body of this function is indented only one space instead of two.

Consider adding this before this function:

```
_dtf: null, // Lazily initialized DateTimeFormat instance
```

Actually, I think it is now guaranteed that the gallery and video apps are always going to be creating a DateTimeFormat object, so maybe there isn't really any need to bother with this lazy initialization.  What if you just had

```
dtf: new navigator.mozL10n.DateTimeFormat()
```
Then just use this.dtf when needed?

Of course, then the order in which the modules were loaded would matter...